### ISSUE TITLE
Race between VM creation and vmWaitCleanup gorutine

### ISSUE THREAD

**Issue by ormergi** [k8snetworkplumbingwg/kubemacpool#143]
**What happened**:
TL:DR If we drop the `Eventually` from [2633 test](https://github.com/k8snetworkplumbingwg/kubemacpool/blob/bd3057c490e011624ba7992af0b4e8d6bf8a5d81/tests/virtual_machines_test.go#L404), its causes a race between VM creation request to vmWaitCleanup gorutine

The [test 2633](https://github.com/k8snetworkplumbingwg/kubemacpool/blob/bd3057c490e011624ba7992af0b4e8d6bf8a5d81/tests/virtual_machines_test.go#L404) first creates a VM by applying a bad yaml (not valid VM create request to kube-api) and expecting to fail, then fixes the yaml and apply it again.

If we do not use `Eventually` on the second `Create` request,
the [gorutine](https://github.com/k8snetworkplumbingwg/kubemacpool/blob/master/pkg/pool-manager/pool.go#L95) that run [vmWaitCleanup](https://github.com/k8snetworkplumbingwg/kubemacpool/blob/bd3057c490e011624ba7992af0b4e8d6bf8a5d81/pkg/pool-manager/virtualmachine_pool.go#L432) in the background (every 3 seconds) wont clear the MAC that requested from the vmWaitConfigMap and internal cache on time and cause the allocation to fail with "MAC already allocated" error, and so is the test.

Logs for test 2633 https://paste.centos.org/view/0050a9a5
Focused on[ first test](https://github.com/k8snetworkplumbingwg/kubemacpool/blob/bd3057c490e011624ba7992af0b4e8d6bf8a5d81/tests/virtual_machines_test.go#L406) `It` https://paste.centos.org/view/abfaa692

**What you expected to happen**:
Successfully create a VM by applying a bad  yaml, then applying another yaml with the same MAC address.

**How to reproduce it (as minimally and precisely as possible)**:
Run test 2633 w/o `Eventually` at https://github.com/k8snetworkplumbingwg/kubemacpool/blob/bd3057c490e011624ba7992af0b4e8d6bf8a5d81/tests/virtual_machines_test.go#L421 and https://github.com/k8snetworkplumbingwg/kubemacpool/blob/bd3057c490e011624ba7992af0b4e8d6bf8a5d81/tests/virtual_machines_test.go#L446
OR
Apply [this]( https://paste.centos.org/view/51674482) yaml (bad VM yaml) and [this](https://paste.centos.org/view/013e6c97) yaml (fixed yaml) right away
Example:
https://imgur.com/xK7ZIfw
**Anything else we need to know?**:

**Environment**:

**Comment by RamLavi** [comment]
This issue should not occur on the new design. closing issue
